server.js

```diff
@@ -0,0 +1,236 @@
+  req.on("data", chunk => {
+    let test = chunk.toString();
+    body += chunk;
+  });
+
+  req.on("end", () => {
+    if (req.method === `POST`) {
+      postHandler(req, res, body);
+    } else if (req.method === `GET`) {
+      getHandler(req, res);
+    } else if (req.method === `PUT`) {
+      putHandler(req, res, body);
+    } else if (req.method === `DELETE`) {
+      deleteHandler(req, res, body);
+    }
+  });
+});
+
+server.listen(PORT, () => {
+  console.log(`Server started on PORT: ${PORT}`);
+});
+
+function getHandler(req, res) {
+  if (req.url === `/` || req.url === `/index.html`) {
+    fs.readFile(`./public/index.html`, function(err, data) {
+      if (err) {
+        errorHandler(res);
+      } else {
+        res.writeHead(200, {
+          "content-type": "text/html",
+          "content-length": data.length
+        });
+        res.write(data.toString());
+        res.end();
+      }
+    });
+  } else if (req.url === `/css/styles.css`) {
+    fs.readFile(`./public/css/styles.css`, function(err, data) {
+      if (err) {
+        errorHandler(res);
+      } else {
+        res.writeHead(200, {
+          "content-type": "text/css",
+          "content-length": data.length
+        });
+        res.write(data.toString());
+        res.end();
+      }
+    });
+  } else {
+    fs.readFile(`./public${req.url}`, function(err, data) {
+      if (err) {
+        errorHandler(res);
+      } else {
+        res.writeHead(200, {
+          "content-type": "text/html",
+          "content-length": data.length
+        });
+        res.write(data.toString());
+        res.end();
+      }
+    });
+  }
+}
+
+function errorHandler(res) {
+  fs.readFile(`./public/404.html`, function(err, data) {
+    if (err) {
+      res.writeHead(500, {
+        "content-type": "application/json"
+      });
+    }
+    res.writeHead(404, {
+      "content-type": "text/html",
+      "content-length": data.length
+    });
+    res.write(data.toString());
+    res.end();
+  });
+}
+
+function postHandler(req, res, body) {
+  let reqParse = querystring.parse(body);
+  fs.writeFile(
+    `./public/${reqParse.elementName}.html`,
+    newPost(reqParse),
+    (err, data) => {
+      if (err) {
+        return errorHandler(res);
+      }
+
+      //rewriting Index
+      fs.readdir(`./public`, function(err, data) {
+        if (err) {
+          errorHandler(res);
+        } else {
+          let filtered = data.filter(function(e) {
+            return ![`.keep`, `404.html`, `css`, `index.html`].includes(e);
+          });
+
+          fs.writeFile(`./public/index.html`, makeIndex(filtered), function(
+            err
+          ) {
+            if (err) {
+              errorHandler(res);
+            }
+          });
+
+          let str = `{ "success" : true }`;
+          res.writeHead(200, {
+            "content-type": "application/json",
+            "content-length": str.length
+          });
+          res.write(str);
+          res.end();
```

Comment:
This section of code (let str ... - res.end)  should be in the fs.writeFile above.